server/controllers/finance/reports/systemUsage/index.js

```diff
@@ -19,68 +19,117 @@ async function document(req, res, next) {
       WHERE DATE(created_at) = DATE(?)
       GROUP BY HOUR(created_at)
     `;
 
     const cashNbr = `
-      SELECT HOUR(created_at)  as 'hour', COUNT(cash.uuid) as 'number'
+      SELECT HOUR(created_at) as 'hour', COUNT(cash.uuid) as 'number',
+        GROUP_CONCAT(DISTINCT user_id ORDER BY user_id) AS users
       FROM cash
       WHERE DATE(created_at) = DATE(?)
       GROUP BY HOUR(created_at)
     `;
 
+    const stockMovementNbr = `
+      SELECT HOUR(created_at) as 'hour', COUNT(DISTINCT document_uuid) as 'number',
+        GROUP_CONCAT(DISTINCT user_id ORDER BY user_id) AS users
+      FROM stock_movement
+      WHERE DATE(created_at) = DATE(?)
+      GROUP BY HOUR(created_at)
+    `;
+
     const [
       patients,
       invoices,
       cashPayments,
+      stockMovements,
     ] = await Promise.all([
       db.exec(patientsNbr, date),
       db.exec(invoicesNbr, date),
       db.exec(cashNbr, date),
+      db.exec(stockMovementNbr, date),
     ]);
 
     const patientMap = _.groupBy(patients, 'hour');
     const cashPaymentMap = _.groupBy(cashPayments, 'hour');
     const invoicesMap = _.groupBy(invoices, 'hour');
+    const stockMovementsMap = _.groupBy(stockMovements, 'hour');
 
     const pivot = [];
 
     for (let i = 0; i < 25; i++) {
       const index = `${i}`;
       const row = {
-        hour : index, patients : 0, invoices : 0, cashPayments : 0,
+        hour : index, patients : 0, invoices : 0, cashPayments : 0, stockMovements : 0, users : 0,
       };
 
+      let aggregateUsers = '';
+
       if (patientMap[index]) {
-        row.patients = patientMap[index][0].number;
+        const { number, users } = patientMap[index][0];
+        row.patients = number;
+        aggregateUsers += users;
       }
 
       if (invoicesMap[index]) {
-        row.invoices = invoicesMap[index][0].number;
+        const { number, users } = invoicesMap[index][0];
+        row.invoices = number;
+        aggregateUsers += users;
       }
 
       if (cashPaymentMap[index]) {
-        row.cashPayments = cashPaymentMap[index][0].number;
+        const { number, users } = cashPaymentMap[index][0];
+        row.cashPayments = number;
+        aggregateUsers += users;
       }
 
+      if (stockMovementsMap[index]) {
+        const { number, users } = stockMovementsMap[index][0];
+        row.stockMovements = number;
+        aggregateUsers += users;
+      }
+
+      // make a unique array of users and count them
+      const users = aggregateUsers
+        .split(',')
+        .sort()
+        .filter((value, idx, arr) => arr.indexOf(value) === idx)
+        .length;
+
+      row.users = users;
+
       pivot.push(row);
     }
 
-    const cleaned = pivot.filter(row => (row.patients + row.invoices + row.cashPayments > 0));
+    const cleaned = pivot.filter(row => (row.patients + row.invoices + row.cashPayments + row.stockMovements > 0));
+
+    // aggregate all relevant
+    const aggregates = cleaned.reduce((agg, row) => {
```

Comment:
I don't understand why you need to do pivot.filter.  Wouldn't the reduce work with the pivot array itself, since the aggregates would just add 0 for all the items in the row that are zero?   Is this an efficiency mod that may not be necessary with this small a number of items from the queries?

Reply:
Filter removes rows that are all zero - it is just a way to clean up rows that are 0 and don't have any data in them.